src/pages/[[...markdownPath]].js

```diff
@@ -216,10 +220,31 @@ export async function getStaticProps(context) {
   const fm = require('gray-matter');
   const meta = fm(mdx).data;
 
+  // Serialize MDX into JSON.
+  const content = JSON.stringify(children, stringifyNodeOnServer);
+  toc = JSON.stringify(toc, stringifyNodeOnServer);
+
+  // Do not cache /error-decoder pages in local filesystem, so we can have
+  // latest error codes from GitHub.
+  if (path === 'error-decoder') {
+    return {
+      props: {
+        content,
+        toc,
+        meta,
+        errorCodes: await (
+          await fetch(
+            'https://raw.githubusercontent.com/facebook/react/main/scripts/error-codes/codes.json'
+          )
+        ).json(),
+      },
+    };
+  }
```

Comment:
The legacy site implements a Gatsby plugin to download the `codes.json` from GitHub RAW during the build: https://github.com/reactjs/reactjs.org/blob/d696cb1d2a1ea1d7eb92d5b098b51dde8fcc536c/plugins/gatsby-source-react-error-codes/gatsby-node.js#L13-L30

I am fetching `codes.json` in `getStaticProps`, but IMHO this doesn't look right.

Reply:
What if we also use `getStaticPaths` to generate all of the error code pages at build time? We should also namespace the route with something like `errors/[[error_code]].js`, so the new site URLs would be something like:

`https://react.dev/errors/377?args%5B0%5D=114514n&args%5B1%5D=count`

That would require a few other things:
- For when new errors are pushed but the site hasn't deployed, we would need a fallback for Not Found that looks up the error codes at runtime to see if they exist.
- For redirects to work (where the error code is in the URL for the static paths) we would need a query-param based redirect. Maybe `https://react.dev/errors?invariant=337` redirects to `https://react.dev/errors/337`. We can update the error messages in React to use the static path and keep the redirect around as a 301 for existing links.

Reply:
We could avoid the later if vercel allows for moving query params to the route in a redirect, but skimming the docs I don't see a way to do it.

Reply:
> What if we also use `getStaticPaths` to generate all of the error code pages at build time? We should also namespace the route with something like `errors/[[error_code]].js`, so the new site URLs would be something like:

I like the idea of `https://react.dev/errors/377?args%5B0%5D=114514n&args%5B1%5D=count`, but SSG all 483 (and the number will continue to grow) pages at the build time really sounds like a bad idea to me.

> For when new errors are pushed but the site hasn't deployed, we would need a fallback for Not Found that looks up the error codes at runtime to see if they exist.

It is possible to implement the on-demand static generation with the Next.js' `getStaticPaths` + `fallback: 'blocking'` (https://nextjs.org/docs/pages/api-reference/functions/get-static-paths#fallback-blocking). However, `getStaticPaths` + `fallback: 'blocking'` doesn't work with Next.js SSG, which means the site now has to run on a server/serverless platform.

Should we fetch the `code.json` on the client side (which can guarantee to be the latest version) instead?

Reply:
What's the concern with SSG? If it's build-time perf, what about using ISR with [`fallback: true`](https://nextjs.org/docs/pages/api-reference/functions/get-static-paths#fallback-true)? This will avoid generating the files during build time and instead generate them on-demand, forcing blocking mode where it makes sense (e.g. Google). Vercel will automatically configure the server for deployment. 

Fetching on the client would create an unnecessary delay/waterfall.

Reply:
Done in [`4c89c2b` (#6214)](https://github.com/reactjs/react.dev/pull/6214/commits/4c89c2beb2373d4db21fdaa301d60c303108c59c). I have created a `src/pages/errors/[error_code].tsx` and a `src/pages/errors/index.tsx` file.